src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -1,123 +1,233 @@
-     * @return
-     */
-    public List<String> getTopThreeWeatherCondition7(){
-        return null;
-    }
-
-    /**
-     * return a multimap where key is a district authority and values are accident ids
-     * ex:
-     * authority1 -> id1, id2, id3
-     * authority2 -> id4, id5
-     * @return
-     */
-    public Multimap<String, String> getAccidentIdsGroupedByAuthority7(){
-        return null;
-    }
-
-
-    // Now let's do same tasks but now with streaming api
-
-
-
-    public RoadAccident getAccidentByIndex(String index){
-        return null;
-    }
-
-
-    /**
-     * filter list by longtitude and latitude fields
-     * @param minLongitude
-     * @param maxLongitude
-     * @param minLatitude
-     * @param maxLatitude
-     * @return
-     */
-    public Collection<RoadAccident> getAccidentsByLocation(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude){
-        return null;
-    }
-
-    /**
-     * find the weather conditions which caused max number of incidents
-     * @return
-     */
-    public List<String> getTopThreeWeatherCondition(){
-        return null;
-    }
-
-    /**
-     * count incidents by road surface conditions
-     * @return
-     */
-    public Map<String, Long> getCountByRoadSurfaceCondition(){
-        return null;
-    }
-
-    /**
-     * To match streaming operations result, return type is a java collection instead of multimap
-     * @return
-     */
-    public Map<String, List<String>> getAccidentIdsGroupedByAuthority(){
-        return null;
-    }
+	private final List<RoadAccident> roadAccidentList;
+
+	public DataProcessor(List<RoadAccident> roadAccidentList) {
+		this.roadAccidentList = roadAccidentList;
+	}
+
+	// First try to solve task using java 7 style for processing collections
+
+	/**
+	 * Return road accident with matching index
+	 * 
+	 * @param index
+	 * @return
+	 */
+	public RoadAccident getAccidentByIndex7(String index) {
+		RoadAccident roadAccident = null;
+		if (roadAccidentList != null && roadAccidentList.size() > 0) {
+			for (RoadAccident accident : roadAccidentList) {
+				if (accident.getAccidentId() != null
+						&& accident.getAccidentId().equals(index)) {
+					roadAccident = accident;
+					break;
+				}
+			}
+		}
+		return roadAccident;
+	}
+
+	/**
+	 * filter list by longtitude and latitude values, including boundaries
+	 * 
+	 * @param minLongitude
+	 * @param maxLongitude
+	 * @param minLatitude
+	 * @param maxLatitude
+	 * @return
+	 */
+	public Collection<RoadAccident> getAccidentsByLocation7(float minLongitude,
+			float maxLongitude, float minLatitude, float maxLatitude) {
+		Collection<RoadAccident> colls = null;
+		if (roadAccidentList != null && roadAccidentList.size() > 0) {
+			colls = new ArrayList<RoadAccident>();
+			for (RoadAccident accident : roadAccidentList) {
+				if (accident.getLongitude() <= maxLongitude
```

Comment:
It's fine to use common method if logic is complex,or direct way like my usage if the logic is simple~